client/state/data-layer/wpcom/activity-log/rewind/restore-status/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { pick, delay } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	REWIND_RESTORE_PROGRESS_REQUEST,
+} from 'state/action-types';
+import {
+	getRewindRestoreProgress,
+	updateRewindRestoreProgress,
+	rewindRestoreUpdateError
+} from 'state/activity-log/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+
+const requestRestoreProgress = ( { dispatch }, action ) => {
+	dispatch( http( {
+		method: 'GET',
+		path: `/activity-log/${ action.siteId }/rewind/${ action.timestamp }/restore-status`,
+		apiVersion: '1',
+	}, action ) );
+};
+
+const fromApi = data => {
+	return {
+		status: data.status,
+		percent: data.percent,
+		message: data.message,
+		errorCode: data.error_code,
+		failureReason: data.failure_reason,
+	};
+};
+
+export const receiveRestoreProgress = ( { dispatch }, { siteId, timestamp }, next, data ) => {
+	const POLLING_DELAY_MS = 1500;
+	dispatch( updateRewindRestoreProgress( fromApi( data ) ) );
+	if ( data.status !== 'finished' ) {
+		delay( dispatch, POLLING_DELAY_MS, getRewindRestoreProgress( siteId, timestamp ) );
+	}
+};
+
+export const receiveRestoreError = ( { dispatch }, { siteId, timestamp }, next, error ) => {
+	dispatch( rewindRestoreUpdateError(
```

Comment:
This will indicate a _network_ error, as opposed to a _restore_ error. We'll need to think about that. What happens if you have restore progress information, but a network request fails?

We currently split restore errors and restore progress into different parts of the state. That doesn't seem to make much sense after some recent conversation (https://github.com/Automattic/wp-calypso/pull/15284#issuecomment-309821710). Perhaps network errors should be stored in the tree at `state.activityLog.restoreError[siteId]`.

Some number of retries with a backoff seems sensible, I think data-layer offers some of this, I'll dig around.

Maybe a notice or smaller banner would be good to show saying we can't check restore progress but we'll keep trying.